include/rapidjson/reader.h

```diff
@@ -955,6 +956,9 @@ class GenericReader {
     // Parse string and generate String event. Different code paths for kParseInsituFlag.
     template<unsigned parseFlags, typename InputStream, typename Handler>
     void ParseString(InputStream& is, Handler& handler, bool isKey = false) {
+        if ((parseFlags & kParseInsituFlag) && (parseFlags & kParseMultiLineStringValueFlag))
+            RAPIDJSON_PARSE_ERROR(kParseErrorInvalidFlagCombination, is.Tell());
```

Comment:
Although I understand this logic might be needed to solve our immediate needs it might make it hard for us to integrate back with Tencent's official repository.
Do we have a work item to revisit this implementation later and build a maintenance plan?